flopy/mf6/utils/codegen/filters.py

```diff
@@ -0,0 +1,568 @@
+import sys
+from enum import Enum
+from keyword import kwlist
+from pathlib import Path
+from pprint import pformat
+from typing import Any, List, Optional
+
+from boltons.iterutils import default_enter, remap
+
+
+def _try_get_enum_value(v: Any) -> Any:
+    """
+    Get the enum's value if the object is an instance
+    of an enumeration, otherwise return it unaltered.
+    """
+    return v.value if isinstance(v, Enum) else v
+
+
+def _get_vars(d: dict) -> dict[str, dict]:
+    vars_ = dict()
+    def visit(p, k, v):
+        if isinstance(v, dict) and "type" in v:
+            vars_[k] = v
+        return True
+    def enter(p, k, v):
+        if isinstance(v, dict) and "type" in v:
+            return (v, False)
+        return default_enter(p, k, v)
+
+    remap(d, enter=enter, visit=visit)
+    return vars_
+
+
+class Filters:
```

Comment:
The generated classes are pretty complicated at the moment. The filters are a temporary adapter. Most if not all of this can go away with flopy4, if not sooner